lib/spike.rb

```diff
@@ -0,0 +1,361 @@
+require 'forwardable'
+require 'json'
+require 'pathname'
+require 'active_support/inflector'
+
+module Spike
+  class Resource
+    attr_reader :path, :segments
+    def initialize(path)
+      @path = path.to_s
+      @segments = @path.split("/").reject(&:empty?)
+    end
+
+    def name
+      s = objects.first
+      if subresource?
+        prefix, resource = objects
+        s = [prefix.singularize, resource].join("_")
+      end
+      singular?? s.singularize : s
+    end
+
+    def singular?
+      segments.last =~ /:.*id$/
+    end
+
+    def subresource?
+      objects.count > 1
+    end
+
+    def objects
+      return @objects if defined?(@objects)
+
+      @objects = segments.reject {|segment| segment.start_with?(":")}
+      if @objects.count > 1
+        @objects = @objects.reject {|resource| resource == "repos"}
+      end
+      @objects
+    end
+  end
+
+  class Routes
+    attr_reader :definition
+    def initialize(json)
+      @definition = JSON.parse(json, object_class: OpenStruct)
+      normalize!
+    end
+
+    private
+
+    def normalize!
+      @definition["original_path"] = definition.path
+      @definition["path"]          = normalized_path
+      @definition["params"]        = normalized_params
+
+      @definition["documentation_url"] = @definition["documentationUrl"]
+      @definition["verb"]              = @definition["method"]
+    end
+
+    def normalized_path
+      path = definition.path.to_s
+      if definition.path.to_s.include?(":id")
+        idx = original_path_segments.index(":id")
+        resource = original_path_segments[idx-1].singularize
+        path = path.gsub(":id", ":#{resource}_id")
+      end
+      path
+    end
+
+    def normalized_params
+      params = (@definition.params || []).reject do |param|
+        param.name == "owner"
+      end
+
+      params.each do |param|
+        if param.name == "id"
+          idx = original_path_segments.index(":id")
+          resource = original_path_segments[idx-1].singularize
+          param["name"] = "#{resource}_id"
+        end
+        if param.name.end_with?("_id") && param.type == "string"
+          param["type"] = "integer"
+        end
+        if param.name == "repo"
+          param["description"] = "A GitHub repository"
+        end
```

Comment:
The routes file has an empty string as the description for this, since it doesn't come from the table of inputs on the developer site, it's inferred from the API endpoint path.